Fix the parameter -HvServer to specify the HvServer if there are more than 1 hyperv server

huijuanzhao opened the pull request:
Fix one same issue in two files: NET-IFUP-IFDOWN.ps1, RELOAD-MODULES.ps1.
Change the parameter -HvServer from $TestLocation to $GlobalConfig.Global.Hyperv.Hosts.ChildNodes[0].ServerName, specify the HvServer if there are more than 1 hyperv servers, or will cause PS command fail. 

Before fix the script NET-IFUP-IFDOWN.ps1:
Test failed in RHEL_8.
Errors in output:
10/15/2019 07:57:50 : [WARN ] Get-IPv4ViaKVP: Unable to create Msvm_ComputerSystem object
10/15/2019 07:58:00 : [ERROR] Wait-ForVMToStartKVP: VM LISAv2-OneVM-2019-UJ75-637067517077-role-0 did not start KVP within timeout period (30)
10/15/2019 07:58:00 : [ERROR] Get-IPv4AndWaitForSSHStart: Unable to get ipv4 from VM LISAv2-OneVM-2019-UJ75-637067517077-role-0 via KVP within timeout period (30)

After fix the script NET-IFUP-IFDOWN.ps1:
RHEL_8 test results:
[LISAv2 Test Results Summary]
Test Run On           : 10/16/2019 03:30:46
VHD Under Test        : D:\RHEL-xxxx-x86_64-GEN2-A.vhdx
Initial Kernel Version:  xxxx.el8.x86_64
Final Kernel Version  : xxxx.el8.x86_64
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:6

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes)
-------------------------------------------------------------------------------------------------------------------------------------------
    1 NETWORK              NET-IFUP-IFDOWN                                                                   PASS                 4.82


After fix the script RELOAD-MODULES.ps1:
[LISAv2 Test Results Summary]
Test Run On           : 10/16/2019 05:17:37
VHD Under Test        : D:\RHEL-xxxx-x86_64-GEN2-A.vhdx
Initial Kernel Version: xxxx.el8.x86_64
Final Kernel Version  : xxxx.el8.x86_64
Total Test Cases      : 2 (2 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:5

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes)
-------------------------------------------------------------------------------------------------------------------------------------------
    1 STRESS               STRESSTEST-RELOAD-LIS-MODULES                                                     PASS                 1.16
    2 CORE                 RELOAD-MODULES-SMP                                                                PASS                 1.34

lizzha commented on Libraries/TestLogs.psm1:
```diff
@@ -37,14 +41,15 @@ Function Write-Log()
-		"WARN"	{$fgColor = "Yellow"; continue}
+		"WARN"	{$fgColor = "DarkYellow"; continue}
 		"ERROR"	{$fgColor = "Red"; continue}
+		"VERBOSE"	{$fgColor = "Yellow"; continue}
```
tab before {

lizzha commented on TestControllers/TestController.psm1:
```diff
@@ -24,9 +24,14 @@
 using Module "..\Libraries\TestReport.psm1"
 
+$here = Split-Path -Parent $MyInvocation.MyCommand.Path
+Import-Module (Join-Path $here "..\Libraries\TestReport") -Force -DisableNameChecking
```
it's not necessary, as TestReport only contains classes, it's imported with "using" statement

lizzha commented on Libraries/TestReport.psm1:
```diff
@@ -375,7 +375,11 @@ Class TestSummary
 
-		Set-Content -Value $strHtml -Path $FilePath -Force | Out-Null
+		if (Test-Path (Split-Path $FilePath)) {
+			Set-Content -Value $strHtml -Path $FilePath -Force | Out-Null
```
it can be saved successfully even if the file doesn't exist, so it's not necessary to test-path

ader1990 replied:
I am checking for the parent path if it exists (I have to for the unit tests to pass)

lizzha commented:
The PR includes change for enable verbose log, move logic in run-lisav2.ps1 to module, add unit test. Better to rename the title to reflect the changes.
Also please run regression test on the test

LiliDeng commented:
By default the verbose is turn on?

LiliDeng commented:
Run  BVT-NET-IFUP-IFDOWN against hyperv, hit below exception 
[01/15/2019 06:50:10] [VERBOSE] echo '******' | sudo -S -s eval "export HOME=`pwd`;bash BVT-NET-IFUP-IFDOWN.sh > BVT-NET-IFUP-IFDOWN.log" is running in background with ID 375 ...
[01/15/2019 06:50:11] [ERROR] EXCEPTION: The term 'Get-IPv4AndWaitForSSHStart' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again BVT-NET-IFUP-IFDOWN

ader1990 commented:
@lizzha @LiliDeng I will remove the code that is not related to Run-LISAv2.ps1 refactor, and focus on this PR only on the module creation and unit testing.

LiliDeng commented:
Please provide result for BVT-NET-IFUP-IFDOWN  @asofron

ader1990 commented:
> Please provide result for BVT-NET-IFUP-IFDOWN @asofron

```
Test Run On           : 01/16/2019 06:13:47
[azure] ARM Image Under Test  : Canonical : UbuntuServer : 18.04-LTS : latest
[azure] Total Test Cases      : 1 (1 Pass, 0 Fail, 0 Abort)
[azure] Total Time (dd:hh:mm) : 0:0:6
[azure] 
[azure]    ID TestCaseName                                                                TestResult TestDuration(in minutes) 
[azure] ---------------------------------------------------------------------------------------------------------------------
[azure]     1 BVT-NET-IFUP-IFDOWN                                                               PASS                 4.38 


```

ader1990 commented:
@LiliDeng BVT-NET-IFUP-IFDOWN against hyperv fails, but not because of this patch. this branch is not rebased against your fix https://github.com/LIS/LISAv2/pull/606

LiliDeng commented:
got it

ader1990 commented:
> got it

I rebased on top of master and hyper-v passes too:
```
[LISAv2 Test Results Summary]
[hyperv] Test Run On           : 01/16/2019 07:15:53
[hyperv] VHD Under Test        : Ubuntu_18.04.1\Ubuntu_18.04.1.vhdx
[hyperv] Total Test Cases      : 1 (1 Pass, 0 Fail, 0 Abort)
[hyperv] Total Time (dd:hh:mm) : 0:0:6
[hyperv] 
[hyperv]    ID TestCaseName                                                                TestResult TestDuration(in minutes) 
[hyperv] ---------------------------------------------------------------------------------------------------------------------
[hyperv]     1 BVT-NET-IFUP-IFDOWN                                                               PASS                 4.21 
[hyperv] 
```

ader1990 commented:
@LiliDeng can you merge this patch, if it is all good from your side? thanks

LiliDeng commented:
I will do a basic check, then @lizzha can help merge

ader1990 commented:
because of PR #594 being merged before , I had to add the parameter to the LISAv2-Framework.psm1, so that the feature is unaffected when this patch is merged

Commit:
Add NetworkManager restart in CreateIfupConfigFile for redhat_8

Fix the NetworkManager restart both for dhcp and static mode

lixuemin2016 committed:
Hyper-V: Update Storage cases for skipping IDE and VHD for gen2 vm (#542)

juhlee-microsoft committed:
Use built-in WALA agent (#556)

Use the latest WALA agent in the system as it has the fix of RDMA.

Cleanup Ubuntu support filter.

huijuanzhao committed:
Add NetworkManager restart in CreateIfupConfigFile for redhat_8 (#557)

Fix the NetworkManager restart both for dhcp and static mode

Commit:
Merge remote-tracking branch 'upstream/master'

Commit:
Merge remote-tracking branch 'upstream/master'

Commit:
Merge remote-tracking branch 'upstream/master'

Commit:
Fix the parameter HvServer to specify the hyperv server if there are
more than one server

Commit:
Merge branch 'dev' of https://github.com/huijuanzhao/LISAv2 into HEAD
